plugins/native-script-example/src/main/java/org/elasticsearch/examples/nativescript/script/PopularityScoreScriptFactory.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.elasticsearch.examples.nativescript.script;
+
+import org.apache.lucene.search.Scorer;
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.xcontent.support.XContentMapValues;
+import org.elasticsearch.index.fielddata.ScriptDocValues;
+import org.elasticsearch.script.AbstractDoubleSearchScript;
+import org.elasticsearch.script.ExecutableScript;
+import org.elasticsearch.script.NativeScriptFactory;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Factory for the script that boosts score of a record based on a value of  the record's field.
+ * <p>
+ * This native script demonstrates how to write native custom scores scripts.
+ */
+public class PopularityScoreScriptFactory implements NativeScriptFactory {
+
+    @Override
+    public ExecutableScript newScript(@Nullable Map<String, Object> params) {
+        String fieldName = params == null ? null : XContentMapValues.nodeStringValue(params.get("field"), null);
+        if (fieldName == null) {
+            throw new IllegalArgumentException("Missing the field parameter");
+        }
+        return new PopularityScoreScript(fieldName);
+    }
+
+    /**
+     * Indicates if document scores may be needed by the produced scripts.
```

Comment:
We have these docs already on the base class. I don't think we should copy/paste that for each example.